diff --git a/Lib/venv/__init__.py b/Lib/venv/__init__.py
--- a/Lib/venv/__init__.py
+++ b/Lib/venv/__init__.py
@@ -397,6 +397,7 @@ class EnvBuilder:
                     dest = os.path.join(binpath, dest)
                     try:
                         shutil.copy2(src, dest)
+                        os.chmod(dest, 0o666)
                     except OSError:
                         logger.warning('Unable to copy %r to %r', src, dest)

